pkg/plugins/secret/dev/dev.go

```diff
@@ -0,0 +1,114 @@
+// Copyright 2021 Nitric Pty Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package secret_service
+
+import (
+	"bufio"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"github.com/google/uuid"
+	"github.com/nitric-dev/membrane/pkg/plugins/secret"
+	"github.com/nitric-dev/membrane/pkg/utils"
+)
+
+const DEFAULT_DIR = ".nitric/secrets/"
+
+type DevSecretService struct {
+	secret.UnimplementedSecretPlugin
+	secDir string
+}
+
+type Object struct {
+	Key      string `storm:"id"`
+	Data     string
+	Versions string
+	Labels   string
+}
+
+func (s *DevSecretService) Put(sec *secret.Secret, val []byte) (*secret.SecretPutResponse, error) {
+	if sec == nil {
+		return nil, fmt.Errorf("provide non-empty secret")
+	}
+	if len(sec.Name) == 0 {
+		return nil, fmt.Errorf("provide non-blank secret name")
+	}
+	if len(val) == 0 {
+		return nil, fmt.Errorf("provide non-blank secret value")
+	}
+
+	var versionId = uuid.New().String()
+
+	//Creates a new file in the form:
+	// DIR/Name_Version.txt
+	file, err := os.Create(s.secDir + "/" + sec.Name + "_" + versionId + ".txt")
+	if err != nil {
+		return nil, fmt.Errorf("error creating secret store: %v", err)
+	}
+	writer := bufio.NewWriter(file)
+	writer.WriteString(string(val))
+	writer.Flush()
+	return &secret.SecretPutResponse{
+		SecretVersion: &secret.SecretVersion{
+			Secret: &secret.Secret{
+				Name: sec.Name,
+			},
+			Version: versionId,
+		},
+	}, nil
+}
+
+func (s *DevSecretService) Access(sv *secret.SecretVersion) (*secret.SecretAccessResponse, error) {
+	if sv.Secret.Name == "" {
+		return nil, fmt.Errorf("provide non-blank name")
+	}
+	if sv.Version == "" {
+		return nil, fmt.Errorf("provide non-blank version")
+	}
+
+	content, err := ioutil.ReadFile(s.secDir + "/" + sv.Secret.Name + "_" + sv.Version + ".txt")
```

Comment:
Suggest moving the file name creation to a function so it can be used for both access and put

Reply:
done